src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.DefaultProxyCredentials.cs

```diff
@@ -42,87 +43,90 @@ public void SetGet_Roundtrips()
             }
         }
 
-        [ActiveIssue(23702, TargetFrameworkMonikers.NetFramework)]
         [ActiveIssue(20010, TargetFrameworkMonikers.Uap)]
-        [OuterLoop] // TODO: Issue #11345
         [Fact]
         public async Task ProxyExplicitlyProvided_DefaultCredentials_Ignored()
         {
-            int port;
-            Task<LoopbackGetRequestHttpProxy.ProxyResult> proxyTask = LoopbackGetRequestHttpProxy.StartAsync(out port, requireAuth: true, expectCreds: true);
-            Uri proxyUrl = new Uri($"http://localhost:{port}");
-
             var rightCreds = new NetworkCredential("rightusername", "rightpassword");
             var wrongCreds = new NetworkCredential("wrongusername", "wrongpassword");
+            string expectCreds = "Basic " + Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes($"{rightCreds.UserName}:{rightCreds.Password}"));
 
-            using (HttpClientHandler handler = CreateHttpClientHandler())
-            using (var client = new HttpClient(handler))
+            await LoopbackServer.CreateServerAsync(async (proxyServer, proxyUrl) =>
             {
-                handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, rightCreds);
-                handler.DefaultProxyCredentials = wrongCreds;
-
-                Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
-                Task<string> responseStringTask = responseTask.ContinueWith(t =>
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
                 {
-                    using (t.Result) return t.Result.Content.ReadAsStringAsync();
-                }, TaskScheduler.Default).Unwrap();
-                await (new Task[] { proxyTask, responseTask, responseStringTask }).WhenAllOrAnyFailed();
+                    handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, rightCreds);
+                    handler.DefaultProxyCredentials = wrongCreds;
 
-                TestHelper.VerifyResponseBody(responseStringTask.Result, responseTask.Result.Content.Headers.ContentMD5, false, null);
-                Assert.Equal(Encoding.ASCII.GetString(proxyTask.Result.ResponseContent), responseStringTask.Result);
-
-                string expectedAuth = $"{rightCreds.UserName}:{rightCreds.Password}";
-                Assert.Equal(expectedAuth, proxyTask.Result.AuthenticationHeaderValue);
-            }
+                    Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
+                    await proxyServer.AcceptConnectionAsync(async connection =>
+                    {
+                        await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        if (connection.GetRequestHeaderValue("Proxy-Authorization") == null)
```

Comment:
This feels weird to me.  Are we expecting Proxy-Authorization on this request or not?  Seems like the test should test for exactly what we expect.

Reply:
Problem I run into is that curl handler sends proxy creds without waiting for 407. 
Other handlers wait for the error response. 
I don't know if there is more elegant way how to solve the behavior difference. 
I can certainly wrap that to some helper function.

Reply:
We should have the test logic check for CurlHandler and do different things based on handler.

I assume the type of creds CurlHandler is sending is Basic?  It can't really send Digest without a challenge first.